editoast/src/generated_data/speed_limit_tags_config.rs

```diff
@@ -0,0 +1,22 @@
+use std::collections::HashMap;
+
+use serde::Deserialize;
+use serde::Serialize;
+
+pub type SpeedLimitTagsConfig = HashMap<String, SpeedLimitTag>;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct SpeedLimitTag {
+    pub name: String,
+    #[serde(default)]
+    pub fallback_list: Vec<String>,
+    pub default_speed: f64,
+}
+
+impl SpeedLimitTag {
+    /// Get the speed-limit tags configuration
+    pub fn load() -> SpeedLimitTagsConfig {
```

Comment:
I expect `SpeedLimitTag::load()` to return me a `SpeedLimitTag`, not a `HashMap<String, SpeedLimitTag>`. My guess is that should have implemented this instead.

```rust
impl SpeedLimitTagsConfig {
  pub fn load() -> SpeedLimitTagsConfig { todo!() }
}
```
... but you can't since `SpeedLimitTagsConfig` is only an alias on `HashMap` and you won't be able to implement something for `HashMap`. I see two options here (but would favor the first one).

# [New type pattern](https://rust-unofficial.github.io/patterns/patterns/behavioural/newtype.html)
Instead of

```rust
pub type SpeedLimitTagsConfig = HashMap<String, SpeedLimitTag>;
```
do this
```rust
pub struct SpeedLimitTagsConfig(HashMap<String, SpeedLimitTag>);
```

the drawback is that you would need to redefine all the functions of `HashMap` you need as a proxy (not necessarily bad as code architecture level, just more boilerplate).

# [Extension trait](http://xion.io/post/code/rust-extension-traits.html)
Since you can implement something if you either define the type or the trait, and in this case, you have an alias on the type so you don't define the type, let's make a trait.
```rust
trait Load {
  fn load() -> Self;
}
pub type SpeedLimitTagsConfig = HashMap<String, SpeedLimitTag>;
impl Load for SpeedLimitTagsConfig {
  fn load() -> Self { todo!() }
}
```
but I'm not really fond of this solution because it extends `HashMap` (although the implementation is constrained to a `K = String, V = SpeedLimitTag`).

Reply:
Done in https://github.com/OpenRailAssociation/osrd/pull/8106/commits/e379f7c4c2816f42cddc3c79360b2d20e330beab